Entwurf/Dokumentation.tex

```diff
@@ -0,0 +1,96 @@
+\documentclass[10pt,a4paper]{article}
+\usepackage[utf8]{inputenc}
+\usepackage[german]{babel}
+\usepackage{graphicx}
+\usepackage[table,x11names]{xcolor}
+
+\begin{document}
+
+Die Datenbank für die Applikation besteht aus vier Tabellen : Group, Reservation, Appointment und Booking.
+Die Struktur der Tabellen ist in Abbildung \ref{fig:ER-Diagramm} dargestellt.
+
+\begin{figure}[ht]		
+	\centering
+	\includegraphics[scale=0.3]{ER-Diagramm}
+	\caption{Datenbankentwurf}
+	\label{fig:ER-Diagramm}	
+	\end{figure}
+
+\begin{table}[h]
+\centering
+\caption{Appointment}
+	\label{tab:Appointment}   
+    \begin{tabular}{| l | l | l | l |}
+    \hline
+    \rowcolor{lightgray} Spalte & Datentyp & Beschreibung  \\ \hline
+    Date  & DATE & Datum des Termins(Primarykey).\\ \hline
+    Activated & BIT & Ob der Termin an den gewünschten Tag deaktiviert/aktiviert wird.\\ \hline
+    Startime & TIME & Beginnzeit des Zeitfensters. \\ \hline
+    Endtime & TIME & Endezeit des Zeitfensters.  \\ \hline
+    Note & VARCHAR(128) & Notizen für Appointment(Nullable). \\ \hline
+    \end{tabular}
+\end{table}
+
+
+In Tabelle \ref{tab:Appointment} wird die Struktur der Tabelle Appointment dargestellt. Die Tabelle Appointment speichert alle Termine. Mit dem Feld "Activated" wird gekennzeichnet, ob ein Termin deaktiviert oder aktiviert ist. Die Zustände "gebucht" und "reserviert" ergeben sich daraus, dass ein entsprechender Eintrag in "Booking" oder "Reservation anwesend ist.
+\begin{itemize}
+\item Es darf nur eine Presentation pro tag geben.
+\item Einen Termin kann mit eine Reservierung oder gar keine ausgemacht werden, 
+\end{itemize}
+
+\begin{table}[h]
+\centering
+	\caption{Reservation}
+	\label{tab:Reservation}
+    \begin{tabular}{| l | l | l |}
+    \hline
+    \rowcolor{lightgray} Spalte & Datentyp & Beschreibung  \\ \hline
+    Groups & INTEGER & Die Gruppe, die den Termin reserviert oder gebucht hat(Primarykey, Foreign key auf Groups.Groupnumber).  \\ \hline
+    Appoinment & DATE & Der Termin, der durch die Gruppe gebucht oder reserviert ist(Foreign key auf Appointment.Date). \\ \hline
+    \end{tabular}
+\end{table}
+In Tabelle \ref{tab:Reservation} wird die Struktur der Tabelle Reservation  dargestellt.  In dieser Tabelle werden sowohl Reservierungen, als auch Buchungen eingetragen. Eine Buchung ist dadurch zu erkennen, dass es zusätzlich einen entsprechenden Eintrag in Booking gibt. Der Grund für diese Art der Darstellung ist die Erleichterung der Integritätsprüfung.
+
+\begin{itemize} 
+\item Terminsreservierung kann nur einen Termin( Appointment) haben.
```

Comment:
Andersum: Jeder Termin kann nur einen Eintrag in Reservierung haben

Reply:
Natürlich kann auch jede Reservierung nur einen Termin haben, aber das ist ja sofort erkennbar, weil es ja nur eine Spalte Appintment gibt und da kommt halt ein Termin rein